From 653f40a8365c9bd1fc7dfb78d4b2906fdeebf7b9 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Wed, 13 Jul 2016 17:20:45 +0800
Subject: [PATCH 0595/2241] dts: a3700: add io-expander under i2c0 node

- Add io-expander(pca9555) as expander0 under i2c0 node;
- Expander0 only uses single chip address 0x22 as <reg>, so set i2c's
  address-cells to 1 and size-cells to 0.

Change-Id: Iee513f8849f6f75250aa083fb19b54e5d160f1ed
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31158
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 986be13..21ca1b7 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -93,6 +93,12 @@
 
 			i2c0: i2c@11000 {
 				status = "okay";
+				expander0: pca9555@22 {
+					compatible = "nxp,pca9555";
+					reg = <0x22>;
+					gpio-controller;
+					#gpio-cells = <2>;
+				};
 			};
 
 			i2c1: i2c@11080 {
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 1825522..2ff5ce4 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -119,6 +119,8 @@
 			i2c0: i2c@11000 {
 				compatible = "marvell,armada-3700-i2c";
 				reg = <0x11000 0x80>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				interrupts = <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>;
 				mrvl,i2c-fast-mode;
 				clocks = <&ngateclk 17>;
@@ -128,6 +130,8 @@
 			i2c1: i2c@11080 {
 				compatible = "marvell,armada-3700-i2c";
 				reg = <0x11080 0x80>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
 				mrvl,i2c-fast-mode;
 				clocks = <&ngateclk 16>;
-- 
2.7.4

